src/Datadog.Trace.ClrProfiler.Managed/ScopeFactory.cs

```diff
@@ -191,33 +192,30 @@ public static Scope CreateDbCommandScope(Tracer tracer, IDbCommand command)
             return scope;
         }
 
-        public static string GetDbType(string commandTypeName)
+        public static string GetDbType(string namespaceName, string commandTypeName)
         {
-            switch (commandTypeName)
-            {
-                case "SqlCommand":
-                    return "sql-server";
-                case "NpgsqlCommand":
-                    return "postgres";
-                case "MySqlCommand":
-                    return "mysql";
-                case "OracleCommand":
-                    return "oracle";
-                case "SqliteCommand":
-                case "SQLiteCommand":
-                    return "sqlite";
-                case "InterceptableDbCommand":
-                case "ProfiledDbCommand":
-                    // don't create spans for these
-                    return null;
-                default:
-                    const string commandSuffix = "Command";
+            const string commandSuffix = "Command";
 
-                    // remove "Command" suffix if present
-                    return commandTypeName.EndsWith(commandSuffix)
-                               ? commandTypeName.Substring(0, commandTypeName.Length - commandSuffix.Length).ToLowerInvariant()
-                               : commandTypeName.ToLowerInvariant();
-            }
+            var result =
+                commandTypeName switch
+                {
+                    "SqlCommand" => "sql-server",
+                    "NpgsqlCommand" => "postgres",
+                    "MySqlCommand" => "mysql",
+                    "SqliteCommand" => "sqlite",
+                    "SQLiteCommand" => "sqlite",
+                    "InterceptableDbCommand" => null,
+                    "ProfiledDbCommand" => null,
+                    _ when namespaceName.Length == 0 && commandTypeName == commandSuffix =>
+                        commandTypeName.ToLowerInvariant(),
```

Comment:
nit: given `comandSuffix` is constant, this will always return `command`, so may as well return `command` directly I think?

```suggestion
                    _ when namespaceName.Length == 0 && commandTypeName == commandSuffix => "command"
```

Reply:
Yes, agreed.